x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -124,14 +123,14 @@ public void validate(ShardFollowTask params, ClusterState clusterState) {
 
     @Override
     public Assignment getAssignment(final ShardFollowTask params, final ClusterState clusterState) {
-        final DiscoveryNode node = selectLeastLoadedNode(
+        final DiscoveryNode selectedNode = selectLeastLoadedNode(
             clusterState,
-            ((Predicate<DiscoveryNode>) DiscoveryNode::isDataNode).and(DiscoveryNode::isRemoteClusterClient)
+            node -> node.isDataNode() && (node.isRemoteClusterClient() || node.getVersion().before(DiscoveryNode.PLUGGABLE_ROLES_VERSION))
```

Comment:
Assigning to a node that is a data node but before version 7.3.0 could still fail (if the node doesn't have the remote cluster client role). I wonder if we should implement this as a last resort? So try to select the least loaded node that is a data node and is a remote cluster client. If that doesn't turn up any nodes, try to assign to a node that is a data node and is before the version that we formalized the remote cluster client role. Note that that is a different version than when we made roles pluggable.

Reply:
I implemented this in https://github.com/elastic/elasticsearch/pull/60093/commits/fda3927f8d904c4fa7ef16f07c87e2b866fcbbb4, but I backed it out because I thought it could lead to hotspots on upgraded nodes. I am okay with either option. I pushed https://github.com/elastic/elasticsearch/pull/60093/commits/7eeaa145c13f8c389e195683335d4e26bf206b7d to restore this behavior.